repo_test.go

```diff
@@ -543,9 +552,17 @@ func (rs *RepoSuite) TestSign(c *C) {
 		s := &data.Signed{}
 		c.Assert(json.Unmarshal(rootJSON, s), IsNil)
 		c.Assert(s.Signatures, HasLen, len(keyIDs))
-		for i, id := range keyIDs {
-			c.Assert(s.Signatures[i].KeyID, Equals, id)
+
+		// Signatures may be in any order, so must sort key IDs before comparison.
+		wantKeyIDs := append([]string{}, keyIDs...)
+		sort.Strings(wantKeyIDs)
+
+		gotKeyIDs := []string{}
+		for _, sig := range s.Signatures {
+			gotKeyIDs = append(gotKeyIDs, sig.KeyID)
 		}
+
+		c.Assert(sortStrings(keyIDs), DeepEquals, sortStrings(gotKeyIDs))
```

Comment:
What are we really testing here? That we sorted strings correctly? So why not just reuse the built-in function for sorting strings in the first place?

Reply:
This tests that keyIDs and gotKeyIDs have the same elements, disregarding order. Previously the test only passed if the key IDs were in the same order.

I had a duplicate sorting of `keyIDs`/`wantKeyIDs`, which was perhaps a cause of confusion -- I've fixed this.

The `sortStrings` helper made a copy of the string slice before sorting (since `sort.Strings` sorts in place), but I removed the helper since it was confusing and only used in this one test.